Tools/ci-run.sh

```diff
@@ -34,7 +32,28 @@ elif [[ $OSTYPE == "darwin"* ]]; then
   export CC="clang -Wno-deprecated-declarations"
   export CXX="clang++ -stdlib=libc++ -Wno-deprecated-declarations"
 else
-  echo "No setup specified for $OSTYPE"
+  echo "Skipping compiler setup: No setup specified for $OSTYPE"
+fi
+
+if [[ $COVERAGE == "1" ]]; then
+  echo "Skip setting up compilation caches"
+elif [[ $OSTYPE == "msys" ]]; then
+  echo "Set up sccache"
+  echo "TODO: Make a soft symlink to sccache"
+else
+  echo "Set up ccache"
+  echo "Make a soft symlinks to ccache"
+
+  # sudo /usr/sbin/update-ccache-symlinks ?
+  cp ccache /usr/local/bin/
+  ln -s ccache /usr/local/bin/gcc
+  ln -s ccache /usr/local/bin/g++
+  ln -s ccache /usr/local/bin/cc
+  ln -s ccache /usr/local/bin/c++
+  ln -s ccache /usr/local/bin/clang
+  ln -s ccache /usr/local/bin/clang++
```

Comment:
The [documentation](https://github.com/hendrikmuhs/ccache-action#usage) does not say that this is necessary. Is it really needed? Does the previous call to `/usr/sbin/update-ccache-symlinks` still work?

Reply:
I actually forgot to remove the comment
Fr what's I've seen it doesn't work and even if it did it's does not come from an installation of the ccache, so I think it's better to make sure thta everything works by doing it ourselves

Reply:
What's about "Is this needed at all?" It's a setup step for ccache ("run ccache via symlink"), and if I didn't miss anything, github action only installs the ccache and und saves it, you should set it up by yourself
I'm pretty sure that this step is necessary

Reply:
Under Linux, this shouldn't be necessary because the action [installs `ccache` using `apt`](https://github.com/hendrikmuhs/ccache-action/blob/53e53642769c4b03a7881a7283314d1f364134ec/src/restore.ts#L66). The `.deb` package automatically runs `/usr/sbin/update-ccache-symlinks` after installation.

In fact, it likely has no effect, because the `ci-run.sh` script is still putting `/usr/lib/ccache` first in the `$PATH`. So the symlinks `update-ccache-symlinks` creates there will be used before anything in `/usr/local/bin/`.

Under macOS, the copying and symlinking probably _**is needed**_, because the Homebrew package the action installs `ccache` from wouldn't be allowed to create a `/usr/lib/ccache` directory or any symlinks inside it. So adding that (nonexistent, on macOS) directory to the `$PATH` wouldn't have any effect.

If the Homebrew package is automatically creating links anywhere, they'd either be in `/usr/local/lib/ccache` or `/opt/homebrew/lib/ccache`, depending on whether it's an Intel or Apple Silicon based system. But the `/usr/local/bin/` symlinks are probably as good as any. The Homebrew package probably wouldn't automatically place the symlink directory in the `$PATH` because it "conflicts with the system compiler", something the Homebrew project is constantly tiptoeing around to avoid incurring Apple's wrath.

Reply:
Windows is the tricky one. (sccache doesn't seem to be working at all there, currently — it never caches anything.)

A note in the `ci-run.sh` script says "TODO: Make a soft symlink to sccache". But mozilla's docs say [symlinks don't work with `sccache`, they have to be hard links](https://github.com/mozilla/sccache#symbolic-links).

I'm not sure how well (or if at all) hardlinks work in MSYS environments. I've used `ccache` on Windows, but only in MSYS2, installed from their packages. There are actually two different versions of how those work, depending on the environment.

The `msys/ccache` package (which is used in MSYS2's Unix-like environment) creates symlinks in `/usr/lib/ccache/bin`, similar to the Debian package (but with an extra `bin/` subdirectory). That directory is automatically placed first in the `$PATH` when running under the MSYS shell.

But the packages for the MSYS2 Windows-compatible environments (MinGW64, Clang64, UCRT, etc.) are different. _Those_ install a directory `lib/ccache/bin/` in the environment space, which is again placed first in the `$PATH` under the environment's shell. But instead of symlinks, it contains a bunch of small scripts named for the various "linked" compilers, each of which invokes that compiler via `ccache`.

For example, this is the `mingw-w64-x86_64-ccache` package's `/mingw64/lib/ccache/bin/g++` script:

```bash
#!/bin/bash
ccache /mingw64/bin/g++ "$@"
```

Something along those lines might be the only good option for running `sccache` effectively on Windows, too.

Reply:
It would be great to get sccache running. The windows ci jobs always take a lot of time. @ferdnyc if you have the time for it, creating a pr to fix this problem would be appreciated.